src/game/engine/anim.cpp

```diff
@@ -134,90 +162,135 @@ void AnimClass::AI()
 #endif
 }
 
+/**
+ * @brief
+ *
+ * @address 0x00425610
+ */
 LayerType AnimClass::In_Which_Layer() const
 {
     // Infantry corpses need to be rendered below everything else.
     if (What_Type() == ANIM_CORPSE1 || What_Type() == ANIM_CORPSE2 || What_Type() == ANIM_CORPSE3) {
         return LAYER_SURFACE;
     }
 
-    if (m_Class->Is_Surface() || Target_Legal(m_AttachedTo)) {
+    if (Target_Legal(m_AttachedTo) || m_Class->Is_Surface()) {
         return LAYER_GROUND;
     }
 
     // All other anim types default to AIR.
     return LAYER_AIR;
 }
 
+/**
+ * @brief
+ *
+ * @address 0x00425A98
+ */
 void AnimClass::Detach(target_t target, int a2)
 {
-#ifdef GAME_DLL
-    void (*func)(AnimClass *, target_t, int) = reinterpret_cast<void (*)(AnimClass *, target_t, int)>(0x00425A98);
-    func(this, target, a2);
-#else
-    DEBUG_ASSERT_PRINT(false, "Unimplemented function called!\n");
-#endif
+    if (target == m_AttachedTo) {
```

Comment:
Combine these into one if such as `if (arget == m_AttachedTo && a2)` if there is nothing else done between the two ifs. Also, a2 looks like its a BOOL, to_detach perhaps?